core/src/common/lwm2m_result.c

```diff
@@ -60,3 +61,30 @@ int AwaLwm2mResult_ToAwaError(AwaLwm2mResult result, int defaultError)
         return defaultError;
     }
 }
+
+int AwaLwm2mResult_ToAwaLwm2mError(AwaLwm2mResult result)
+{
+    AwaLWM2MError error = AwaLWM2MError_Unspecified;
+    switch (result)
+    {
+    case AwaLwm2mResult_Success:
+    case AwaLwm2mResult_SuccessChanged:
+    case AwaLwm2mResult_SuccessContent:
+    case AwaLwm2mResult_SuccessCreated:
+    case AwaLwm2mResult_SuccessDeleted:
+        return AwaLWM2MError_Success;
+    case AwaLwm2mResult_BadRequest:
+        return AwaLWM2MError_BadRequest;
+    case AwaLwm2mResult_Unauthorized:
+        return AwaLWM2MError_Unauthorized;
+    case AwaLwm2mResult_NotFound:
```

Comment:
I Agree that we need to combine these error codes. At least they are all defined in one place now though :)